fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/DataCachePopulateRewriteRule.java

```diff
@@ -0,0 +1,184 @@
+            PhysicalScanOperator scanOperator = (PhysicalScanOperator) optExpression.getOp();
+
+            switch (populateMode) {
+                case NEVER:
+                    return rewritePhysicalScanOperator(scanOperator, false);
+                case ALWAYS:
+                    return rewritePhysicalScanOperator(scanOperator, true);
+                default: {
+                    Table table = scanOperator.getTable();
+
+                    if (!isValidScanOperatorType(scanOperator.getOpType())) {
+                        return rewritePhysicalScanOperator(scanOperator, false);
+                    }
+
+                    // ignore full table scan
+                    if (checkIsFullColumnScan(table, scanOperator)) {
+                        return rewritePhysicalScanOperator(scanOperator, false);
+                    }
+
+                    ScanOperatorPredicates predicates = null;
+                    try {
+                        // ScanOperatorPredicates maybe nullptr, we have to check it
+                        predicates = scanOperator.getScanOperatorPredicates();
+                    } catch (AnalysisException e) {
+                        LOG.warn("Failed to get ScanOperatorPredicates", e);
+                    }
+
+                    if (predicates == null) {
+                        LOG.warn("ScanOperatorPredicates can't be null");
+                        return rewritePhysicalScanOperator(scanOperator, false);
+                    }
+
+                    // ignore full partition scan
+                    if (checkIsFullPartitionScan(predicates)) {
+                        return rewritePhysicalScanOperator(scanOperator, false);
+                    }
+
+                    return rewritePhysicalScanOperator(scanOperator, true);
+                }
+            }
+        }
+
+        private boolean isValidScanOperatorType(OperatorType operatorType) {
+            if (operatorType == OperatorType.PHYSICAL_HIVE_SCAN || operatorType == OperatorType.PHYSICAL_ICEBERG_SCAN ||
+                    operatorType == OperatorType.PHYSICAL_FILE_SCAN ||
+                    operatorType == OperatorType.PHYSICAL_HUDI_SCAN ||
+                    operatorType == OperatorType.PHYSICAL_DELTALAKE_SCAN ||
+                    operatorType == OperatorType.PHYSICAL_PAIMON_SCAN) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        private boolean checkIsFullColumnScan(Table table, PhysicalScanOperator scanOperator) {
+            int totalColumns = table.getColumns().size();
+            // If it has only one column, ignore check
+            if (totalColumns == 1) {
+                return false;
+            }
+            int usedColumns = scanOperator.getUsedColumns().size();
+            return usedColumns == totalColumns;
+        }
+
+        private boolean checkIsFullPartitionScan(ScanOperatorPredicates scanOperatorPredicates) {
```

Comment:
what if it's a small dimension table?

Reply:
just ignore it, if we introduce a threshold parameter to define what is a small dimension table, it will require users to understand new concepts.

Anyway, the DataCache's space can always be filled, even if we ignore the small dimension tables.

Reply:
I worried that dimension tables would introduce non-negligible IO latency if we didn't cache it

Reply:
So what's your idea?

Introduce a new parameter to define what's is a low dimension table or just didn't check `select *` anymore?